public/app/core/logs_model.ts

```diff
@@ -326,3 +325,141 @@ export function makeSeriesForLogs(rows: LogRowModel[], intervalMs: number): Time
     };
   });
 }
+
+export function seriesDataToLogsModel(seriesData: SeriesData[], intervalMs: number): LogsModel {
+  const metricSeries: SeriesData[] = [];
+  const logSeries: SeriesData[] = [];
+
+  for (const series of seriesData) {
+    if (series.name !== 'logs') {
```

Comment:
I guess we can skip this check since we're validating that a time and message column exists?

Reply:
no need for this -- it just makes it harder for other datasources to implement

Reply:
Removed